edgexfoundry/device-sdk-go
# feat!: change device definition file to YAML format

BREAKING CHANGE: stop supporting TOML format and support YAML format for Device definition files

closes #1380







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1. Run `device-simple` from this branch
2. Verify bootstrap log includes something similiar:
```shell
level=INFO ts=2023-03-15T13:55:19.356608Z app=device-simple source=devices.go:51 msg="Loading pre-defined devices from /Users/chris/Workspace/EdgeX/device-sdk-go/example/cmd/device-simple/res/devices(2 files found)"
level=INFO ts=2023-03-15T13:55:19.35703Z app=device-simple source=devices.go:92 msg="Device Simple-Device01 not found in Metadata, adding it ..."
level=DEBUG ts=2023-03-15T13:55:19.359125Z app=device-simple source=validation.go:58 msg="Device validation request received on message queue. Topic: edgex/device-simple/validate/device, Correlation-id: fd4f7a7d-3c4d-41f2-a46b-c27a345b00eb"
level=DEBUG ts=2023-03-15T13:55:19.361643Z app=device-simple source=validation.go:100 msg="Device validation response published on message queue. Topic: edgex/response/device-simple/fe4aa07d-893b-4c56-95f3-62b650f8fcfc, Correlation-id: fd4f7a7d-3c4d-41f2-a46b-c27a345b00eb"
level=DEBUG ts=2023-03-15T13:55:19.363336Z app=device-simple source=callback.go:58 msg="System event received on message queue. Topic: edgex/system-events/core-metadata/device/add/device-simple/Simple-Device, Correlation-id: dd023c8a-52bd-4860-976a-fb00e1be2a2e"
level=DEBUG ts=2023-03-15T13:55:19.365943Z app=device-simple source=callback.go:70 msg="device Simple-Device01 added"
level=DEBUG ts=2023-03-15T13:55:19.365986Z app=device-simple source=simpledriver.go:292 msg="a new Device is added: Simple-Device01"
level=DEBUG ts=2023-03-15T13:55:19.366078Z app=device-simple source=callback.go:75 msg="Invoked driver.AddDevice callback for Simple-Device01"
level=DEBUG ts=2023-03-15T13:55:19.366129Z app=device-simple source=callback.go:81 msg="starting AutoEvents for device Simple-Device01"
```

## New Dependency Instructions (If applicable)
